src/main/kotlin/org/rust/ide/refactoring/extractTrait/RsExtractTraitProcessor.kt

```diff
@@ -155,6 +156,16 @@ class RsExtractTraitProcessor(
         return impl.parent.addAfter(newImpl, impl) as? RsImplItem
     }
 
+    private fun getTraitVisibility(): String {
+        val visibilities = when (traitOrImpl) {
+            is RsTraitItem -> listOf(traitOrImpl.visibility)
+            is RsImplItem -> members.map { it.visibility }
+            else -> return ""
+        }
+        val unitedVisibility = visibilities.reduce(RsVisibility::unite)
+        return unitedVisibility.format()
+    }
```

Comment:
[minor] this might be more clear
```suggestion
    private fun getTraitVisibility(): String {
        val unitedVisibility = when (traitOrImpl) {
            is RsTraitItem -> traitOrImpl.visibility
            is RsImplItem -> members.map { it.visibility }.reduce(RsVisibility::unite)
            else -> error("unreachable")
        }
        return unitedVisibility.format()
    }
```